src/map.react.js

```diff
@@ -303,6 +305,20 @@ export default class MapGL extends Component {
     map.addSource(update.id, newSource);
   }
 
+  // Hover and click only query layers whose interactive property is true
+  // If no interactivity is specified, query all layers
+  _updateQueryParams(mapStyle) {
+    if (mapStyle instanceof Immutable.Map) {
```

Comment:
Might want to use `Map#isMap` since instanceof checks can be slow unless you specifically want to support classes that inherit from Immutable.Map.

http://facebook.github.io/immutable-js/docs/#/Map/isMap